api/handler/multipart_upload.go

```diff
@@ -0,0 +1,536 @@
+		Bucket             string            `xml:"Bucket"`
+		CommonPrefixes     []CommonPrefix    `xml:"CommonPrefixes"`
+		Delimiter          string            `xml:"Delimiter,omitempty"`
+		EncodingType       string            `xml:"EncodingType,omitempty"`
+		IsTruncated        bool              `xml:"IsTruncated"`
+		KeyMarker          string            `xml:"KeyMarker"`
+		MaxUploads         int               `xml:"MaxUploads"`
+		NextKetMarker      string            `xml:"NextKetMarker,omitempty"`
+		NextUploadIdMarker string            `xml:"NextUploadIdMarker,omitempty"`
+		Prefix             string            `xml:"Prefix"`
+		Uploads            []MultipartUpload `xml:"Upload"`
+		UploadIdMarker     string            `xml:"UploadIdMarker,omitempty"`
+	}
+
+	ListPartsResponse struct {
+		XMLName              xml.Name      `xml:"http://s3.amazonaws.com/doc/2006-03-01/ ListPartsresult" json:"-"`
+		Bucket               string        `xml:"Bucket"`
+		Initiator            Initiator     `xml:"Initiator"`
+		IsTruncated          bool          `xml:"IsTruncated"`
+		Key                  string        `xml:"Key"`
+		MaxParts             int           `xml:"MaxParts,omitempty"`
+		NextpartNumberMarker int           `xml:"NextpartNumberMarker,omitempty"`
+		Owner                Owner         `xml:"Owner"`
+		Parts                []*layer.Part `xml:"Part"`
+		PartNumberMarker     int           `xml:"PartNumberMarker,omitempty"`
+		StorageClass         string        `xml:"StorageClass,omitempty"`
+		UploadId             string        `xml:"UploadId"`
+	}
+
+	MultipartUpload struct {
+		Initiated    string    `xml:"Initiated"`
+		Initiator    Initiator `xml:"Initiator"`
+		Key          string    `xml:"Key"`
+		Owner        Owner     `xml:"Owner"`
+		StorageClass string    `xml:"StorageClass,omitempty"`
+		UploadID     string    `xml:"UploadId"`
+	}
+
+	Initiator struct {
+		ID          string `xml:"ID"`
+		DisplayName string `xml:"DisplayName"`
+	}
+
+	CompleteMultipartUpload struct {
+		XMLName xml.Name               `xml:"http://s3.amazonaws.com/doc/2006-03-01/ CompleteMultipartUpload"`
+		Parts   []*layer.CompletedPart `xml:"Part"`
+	}
+)
+
+const (
+	uploadIdHeaderName   = "uploadId"
+	partNumberHeaderName = "partNumber"
+	maxSizeUploadsList   = 1000
+	maxSizePartsList     = 1000
+	uploadMaxPartNumber  = 10000
+)
+
+func (h *handler) CreateMultipartUploadHandler(w http.ResponseWriter, r *http.Request) {
+	reqInfo := api.GetReqInfo(r.Context())
+
+	bktInfo, err := h.obj.GetBucketInfo(r.Context(), reqInfo.BucketName)
+	if err != nil {
+		h.logAndSendError(w, "could not get bucket info", reqInfo, err)
+		return
+	}
+
+	if err := checkOwner(bktInfo, r.Header.Get(api.AmzExpectedBucketOwner)); err != nil {
+		h.logAndSendError(w, "expected owner doesn't match", reqInfo, err)
+		return
+	}
+
+	metadata := parseMetadata(r)
+	uploadID := uuid.New()
+
+	additional := []zap.Field{zap.String("uploadID", uploadID.String()), zap.String("Key", reqInfo.ObjectName)}
```

Comment:
```suggestion
	additional := []zap.Field{zap.String("uploadID", uploadID.String()), zap.String("key", reqInfo.ObjectName)}
```

And down there again.